### PR TITLE
fix(eslint-plugin-turbo) allow comments in turbo.json file

### PR DESCRIPTION
**Pull request by jooshua-inglis**
The `eslint-plugin-turbo` reads the workspace `turbo.json` file for the `no-undeclared-env-vars` rule. It used the built in javascript json parser to read the config file, which doesn't support comments.

This PR alteres this to use [JSON5](https://www.npmjs.com/package/json5) to parse the `turbo.json` file.

Related to #2796

### CODE REVIEW

**Comment by jooshua-inglis** [comment]
Closing because already fixed

**Comment by jooshua-inglis** [comment]
Sorry, didn't mean to close

File: packages/eslint-plugin-turbo/package.json [hunk 1]
```diff
@@ -34,11 +34,12 @@
     "@types/jest": "^27.4.0",
     "@types/node": "^16.11.12",
     "jest": "^27.4.3",
+    "json5": "^2.2.1",
     "ts-jest": "^27.1.1",
     "tsconfig": "workspace:*",
     "tsup": "^6.2.0",
-    "turbo-utils": "workspace:*",
     "turbo-types": "workspace:*",
+    "turbo-utils": "workspace:*",
```
Comment (jooshua-inglis) [inline packages/eslint-plugin-turbo/package.json:42]: pnpm reordered these automatically

**Comment by tknickman** [comment]
Thanks @jooshua-inglis, can you add tests?

**Comment by jooshua-inglis** [comment]
> Thanks @jooshua-inglis, can you add tests?

Sure thing

**Review by tknickman — APPROVED** [review]

**Pull request merged**